internal/inspect/policy.go

```diff
@@ -21,16 +21,18 @@ import (
 
 func Policies() *Policy {
 	return &Policy{
-		toResolve: make(map[string]map[string]bool),
-		imports:   make(map[string][]string),
-		results:   make(map[string]*responsev1.InspectPoliciesResponse_Result),
+		derivedRolesToResolve: make(map[string]map[string]bool),
+		variablesToResolve:    make(map[string]map[string]bool),
+		imports:               make(map[string][]string),
```

Comment:
I think it would be cleaner to have two separate fields for derived roles imports and variables imports. You don't have to do extra type checks later on because you know exactly what kind they are supposed to be.